src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,268 @@
+
+### POST Request
+To add a new task, implement a route that handles POST requests:
+```python
+@app.route("/tasks", methods=["POST"])
+def create_task():
+    new_task = request.get_json()
+    tasks.append(new_task)
+    return jsonify({"task": new_task}), 201
+```
+`request.get_json()` retrieves the JSON data from the request body and adds the new task to the list of tasks. After the new task has been added to the list of tasks it will return the newly created task as a JSON response with `201` status code, indicating that the task was successfully created.
+
+### PUT Request
+To update a task, implement a route that handles PUT requests:
+```python
+@app.route("/tasks/<int:task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    task.update(request.get_json())
+    return jsonify({"task": task})
+```
+We use the id of the task that we want to update. First, it will look whether the task with provided id exists or not. If the task doesn't exist then it will return the error in JSON response indicating the task was not found, with status code `404`. If the task exists then `task.update(request.get_json())` will update the task with the new data from the request body and returns the updated task as a JSON response.
+
+### DELETE Request
+To delete a task, implement a route that handles DELETE requests:
+```python
+@app.route("/tasks/<int:task_id>", methods=["DELETE"])
+def delete_tasks(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    tasks.remove(task)
+    return jsonify({"result": "Task deleted"})
+```
+We use the id of the task that we want to delete. First, it will look whether the task with provided id exists or not. If the task doesn't exist then it will return the error in JSON response indicating the task was not found, with status code `404`. If the task exists then it will remove that specific task from the list of task and returns a JSON repsonse indicating task was deleted successfully.
+
+### Running the Application
+To run your application, add the following line:
+```python
+if __name__ == "__main__":
+    app.run(debug=True)
+```
+
+Then open your terminal and run the following command form your project directory:
+```bash
+python3 app.py
+```
+Your REST API is now up and running on `http://127.0.0.1:5000`
+
+## Testing your API
+You can test your API using tools like Postman or cURL.
+
+### List all tasks
+You can use the following curl command to list all the avaliable tasks:
+```bash
+curl http://127.0.0.1:5000/tasks
+```
+Expected Result:
+```json
+{
+  "tasks": [
+    {
+      "completed": false,
+      "id": 1,
+      "title": "Buy groceries"
+    },
+    {
+      "completed": true,
+      "id": 2,
+      "title": "Clean the house"
+    }
+  ]
+}
```

Comment:
```suggestion
{
  "tasks": [
    {
      "id": 1,
      "title": "Buy groceries"
      "completed": false,
    },
    {
      "id": 2,
      "title": "Clean the house"
      "completed": true,
    }
  ]
}
```

Doesn't the order in which the tasks are returned will be the same as the order in which they are defined in the tasks list

Reply:
This is because of the `jsonify` function which doesn't maintain the order of the keys of the original dictionary. If we want to maintain the order we need extra codes and since, this is just the basics for the REST API, I think adding extra codes to maintain the order of the keys in response won't be necessary.